File: src/minigames/pillar_battle/java/work/lclpnet/ap2/game/pillar_battle/PillarBattleMiniGame.java
package work.lclpnet.ap2.game.pillar_battle;

import net.minecraft.core.RegistryAccess;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import org.jetbrains.annotations.NotNull;
import work.lclpnet.ap2.ApConstants;
import work.lclpnet.ap2.api.game.*;
import work.lclpnet.kibu.translate.text.LocalizedFormat;

public class PillarBattleMiniGame implements MiniGame {

    @Override
    public boolean canBeFinale(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public @NotNull ResourceLocation getId() {
        return ApConstants.identifier("pillar_battle");
    }

    @Override
    public @NotNull GameType getType() {
        return GameType.FFA;
    }

    @Override
    public @NotNull String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public @NotNull ItemStack getIcon(@NotNull RegistryAccess manager) {
        return new ItemStack(Items.PURPUR_PILLAR);
    }

    @Override
    public Object[] getDescriptionArguments() {
        return new Object[] {LocalizedFormat.format("%.1f", PillarBattleInstance.RANDOM_ITEM_DELAY_TICKS / 20f)};
    }

    @Override
    public @NotNull MiniGameInstance createInstance(@NotNull MiniGameHandle gameHandle) {
        return new PillarBattleInstance(gameHandle);
    }
}

File: src/minigames/pillar_battle/src/main/kotlin/work/lclpnet/ap2/game/pillar_battle/PillarBattleMiniGame.kt
package work.lclpnet.ap2.game.pillar_battle

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.game.GameStartContext
import work.lclpnet.ap2.api.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.MiniGameHandle
import work.lclpnet.ap2.game.MiniGameInstance
import work.lclpnet.kibu.translate.text.LocalizedFormat

class PillarBattleMiniGame : MiniGame {
    override val id = ApConstants.identifier("pillar_battle")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess): ItemStack = ItemStack(Items.PURPUR_PILLAR)
    override fun canBeFinale(context: GameStartContext): Boolean = true
    override fun canBePlayed(context: GameStartContext): Boolean = true
    override val descriptionArguments: Array<Any> = arrayOf(LocalizedFormat.format("%.1f", RANDOM_ITEM_DELAY_TICKS / 20f))
    override fun createInstance(gameHandle: MiniGameHandle): MiniGameInstance = PillarBattleInstance(gameHandle)
}
